app/code/Magento/Backend/Model/Url.php

```diff
@@ -166,7 +167,10 @@ protected function _isSecure()
         if ($this->hasData('secure_is_forced')) {
             return $this->getData('secure');
         }
-        return $this->_scopeConfig->isSetFlag('web/secure/use_in_adminhtml');
+        if ($this->_scopeConfig->isSetFlag('web/secure/use_in_adminhtml')) {
+            return $this->getData('secure');
```

Comment:
Here it must return `true`, note the previous line. It looks like for your particular case you can use the same `secure_is_forced` flag.